tasks/eth/base-002-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: MIT
```

Comment:
@sbvegan could we provide a diff between this upgrade and the OP one here?

Reply:
```
diff NestedSignFromJson.s.sol ../009-fp-upgrade/NestedSignFromJson.s.sol
20d19
< import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
28d26
< import {stdToml} from "forge-std/StdToml.sol";
33,37d30
<     // Safe contract for this task.
<     GnosisSafe securityCouncilSafe = GnosisSafe(payable(vm.envAddress("COUNCIL_SAFE")));
<     GnosisSafe fndSafe = GnosisSafe(payable(vm.envAddress("FOUNDATION_SAFE")));
<     GnosisSafe ownerSafe = GnosisSafe(payable(vm.envAddress("OWNER_SAFE")));
< 
40,41d32
<     
<     address constant optimismPortalGuardian = 0x09f7150D8c019BeF34450d6920f6B3608ceFdAf2;
43,44c34,35
<     /// @notice Verify against https://github.com/ethereum-optimism/superchain-registry/blob/21506ecedf6e83410d12c7cc406685ac061a2a74/superchain/configs/mainnet/base.toml#L44
<     address constant p2pSequencerAddress = 0xAf6E19BE0F9cE7f8afd49a1824851023A8249e8a;
---
>     /// @notice Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/deploy-config/mainnet.json#L12
>     address constant p2pSequencerAddress = 0xAAAA45d9549EDA09E70937013520214382Ffc4A2;
46,47c37,38
<     /// @notice Verify against https://github.com/ethereum-optimism/superchain-registry/blob/21506ecedf6e83410d12c7cc406685ac061a2a74/superchain/configs/mainnet/base.toml#L9
<     address constant batchInboxAddress = 0xFf00000000000000000000000000000000008453;
---
>     /// @notice Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/deploy-config/mainnet.json#L13
>     address constant batchInboxAddress = 0xFF00000000000000000000000000000000000010;
49,50c40,41
<     /// @notice Verify against https://github.com/ethereum-optimism/superchain-registry/blob/21506ecedf6e83410d12c7cc406685ac061a2a74/superchain/configs/mainnet/base.toml#L33
<     address constant batchSenderAddress = 0x5050F69a9786F081509234F1a7F4684b5E5b76C9;
---
>     /// @notice Verify against https://docs.optimism.io/chain/security/privileged-roles#batcher
>     address constant batchSenderAddress = 0x6887246668a3b87F54DeB3b94Ba47a6f63F32985;
52,53c43,44
<     /// @notice Verify against https://github.com/ethereum-optimism/superchain-registry/blob/21506ecedf6e83410d12c7cc406685ac061a2a74/superchain/configs/mainnet/base.toml#L39
<     address constant systemConfigOwner = 0x14536667Cd30e52C0b458BaACcB9faDA7046E056;
---
>     /// @notice Verify against lib/superchain-registry/superchain/extra/addresses/mainnet/op.json
>     address constant systemConfigOwner = 0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A;
55,56c46,47
<     /// @notice Verify onchain
<     uint256 constant l2GenesisBlockGasLimit = 0xABA9500;
---
>     /// @notice Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/deploy-config/mainnet.json#L35
>     uint256 constant l2GenesisBlockGasLimit = 0x1c9c380;
58c49
<     /// @notice Verify onchain
---
>     /// @notice Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/deploy-config/mainnet.json#L37
61c52
<     /// @notice Verify with `cast call 0x73a79Fab69143498Ed3712e519A88a918e1f4072 "scalar()(bytes32)" --rpc-url https://ethereum-rpc.publicnode.com`. We don't link to
---
>     /// @notice Verify with `cast call 0x229047fed2591dbec1eF1118d64F7aF3dB9EB290 "scalar()(bytes32)"`. We don't link to
65c56
<     uint256 constant gasPriceOracleScalar = 0x01000000000000000000000000000000000000000000000000101c12000008dd;
---
>     uint256 constant gasPriceOracleScalar = 0x10000000000000000000000000000000000000000000000000c5fc500000558;
67,68c58,59
<     /// @notice Verify against
<     uint256 constant systemConfigStartBlock = 17482144;
---
>     /// @notice Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/deploy-config/mainnet.json#L44
>     uint256 constant systemConfigStartBlock = 17422444;
70,71c61,62
<     // Verify against the `DisputeGameFactoryProxy` deployment address
<     address constant dgfProxy = 0x43edB88C4B80fDD2AdFF2412A7BebF9dF42cB40e;
---
>     // Verify against the `DisputeGameFactoryProxy` in the Fault Proofs governance post - https://gov.optimism.io/t/upgrade-proposal-fault-proofs/8161
>     address constant dgfProxy = 0xe5965Ab5962eDc7477C8520243A95517CD252fA9;
97c88
<     /// @notice Reads the contract addresses from lib/superchain-registry/superchain/configs/mainnet/base.toml
---
>     /// @notice Reads the contract addresses from lib/superchain-registry/superchain/extra/addresses/mainnet/op.json
99c90
<         string memory addressesToml;
---
>         string memory addressesJson;
103c94
<             string.concat(vm.projectRoot(), "/lib/superchain-registry/superchain/configs/mainnet/base.toml")
---
>             string.concat(vm.projectRoot(), "/lib/superchain-registry/superchain/extra/addresses/mainnet/op.json")
105c96
<             addressesToml = data;
---
>             addressesJson = data;
107c98
<             revert("Failed to read lib/superchain-registry/superchain/configs/mainnet/base.toml");
---
>             revert("Failed to read lib/superchain-registry/superchain/extra/addresses/mainnet/op.json");
110,111c101,102
<         _proxies.L1CrossDomainMessenger = stdToml.readAddress(addressesToml, "$.addresses.L1CrossDomainMessengerProxy");
<         _proxies.L1StandardBridge = stdToml.readAddress(addressesToml, "$.addresses.L1StandardBridgeProxy");
---
>         _proxies.L1CrossDomainMessenger = stdJson.readAddress(addressesJson, "$.L1CrossDomainMessengerProxy");
>         _proxies.L1StandardBridge = stdJson.readAddress(addressesJson, "$.L1StandardBridgeProxy");
113,117c104,108
<             stdToml.readAddress(addressesToml, "$.addresses.OptimismMintableERC20FactoryProxy");
<         _proxies.OptimismPortal = stdToml.readAddress(addressesToml, "$.addresses.OptimismPortalProxy");
<         _proxies.OptimismPortal2 = stdToml.readAddress(addressesToml, "$.addresses.OptimismPortalProxy");
<         _proxies.SystemConfig = stdToml.readAddress(addressesToml, "$.addresses.SystemConfigProxy");
<         _proxies.L1ERC721Bridge = stdToml.readAddress(addressesToml, "$.addresses.L1ERC721BridgeProxy");
---
>             stdJson.readAddress(addressesJson, "$.OptimismMintableERC20FactoryProxy");
>         _proxies.OptimismPortal = stdJson.readAddress(addressesJson, "$.OptimismPortalProxy");
>         _proxies.OptimismPortal2 = stdJson.readAddress(addressesJson, "$.OptimismPortalProxy");
>         _proxies.SystemConfig = stdJson.readAddress(addressesJson, "$.SystemConfigProxy");
>         _proxies.L1ERC721Bridge = stdJson.readAddress(addressesJson, "$.L1ERC721BridgeProxy");
120,128c111,116
<         // Read superchain.toml
<         string memory chainConfig;
<         string memory path =
<             string.concat("/lib/superchain-registry/superchain/configs/mainnet/superchain.toml");
<         try vm.readFile(string.concat(vm.projectRoot(), path)) returns (string memory data) {
<             chainConfig = data;
<         } catch {
<             revert(string.concat("Failed to read ", path));
<         }
---
>         // Read superchain.yaml
>         string[] memory inputs = new string[](4);
>         inputs[0] = "yq";
>         inputs[1] = "-o";
>         inputs[2] = "json";
>         inputs[3] = "lib/superchain-registry/superchain/configs/mainnet/superchain.yaml";
130,132c118
<         _proxies.ProtocolVersions = stdToml.readAddress(chainConfig, "$.protocol_versions_addr");
<         _proxies.SuperchainConfig = stdToml.readAddress(chainConfig, "$.superchain_config_addr");
<     }
---
>         addressesJson = string(vm.ffi(inputs));
134,141c120,121
<     function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
<         allowed = new address[](6);
<         allowed[0] = address(proxies.OptimismPortal2);
<         allowed[1] = address(proxies.SystemConfig);
<         allowed[2] = address(ownerSafe);
<         allowed[3] = address(securityCouncilSafe);
<         allowed[4] = address(fndSafe);
<         // allowed[4] = expectedLivenessGuard;
---
>         _proxies.ProtocolVersions = stdJson.readAddress(addressesJson, "$.protocol_versions_addr");
>         _proxies.SuperchainConfig = stdJson.readAddress(addressesJson, "$.superchain_config_addr");
207c187
<         require(configToCheck.owner() == systemConfigOwner, "300");
---
>         require(configToCheck.owner() == superchainConfigGuardian, "300");
281c261
<         require(portalToCheck.guardian() == optimismPortalGuardian, "6000");
---
>         require(portalToCheck.guardian() == superchainConfigGuardian, "6000");
```